dashboard/src/repositories/postgres/subscription.ts

```diff
@@ -0,0 +1,118 @@
+import prisma from '@/lib/postgres';
+import { Subscription, SubscriptionSchema } from '@/entities/billing';
+
+export async function getUserSubscription(userId: string): Promise<Subscription | null> {
+  try {
+    let subscription = await prisma.subscription.findUnique({
+      where: { userId },
+    });
+
+    if (!subscription) {
+      subscription = await createDefaultStarterSubscription(userId);
+    }
+
+    return SubscriptionSchema.parse(subscription);
+  } catch (error) {
+    console.error('Failed to get user subscription:', error);
+    return null;
+  }
+}
+
+async function createDefaultStarterSubscription(userId: string): Promise<Subscription> {
+  const now = new Date();
+  const periodEnd = new Date(now.getTime() + 30 * 24 * 60 * 60 * 1000);
```

Comment:
Future: We should probably let a time-library handling this, since the avg. month is slightly longer than 30 days (closer to 30.5 days)